mesa/experimental/jupyter_viz.py

```diff
@@ -83,83 +112,98 @@ def handle_change_model_params(name: str, value: any):
                 make_plot(model, measure)
 
 
-@solara.component
-def ModelController(
-    model, play_interval, current_step, set_current_step, reset_counter
-):
+def BaseControls(play_interval, on_step, on_reset):
     playing = solara.use_reactive(False)
-    thread = solara.use_reactive(None)
-    # We track the previous step to detect if user resets the model via
-    # clicking the reset button or changing the parameters. If previous_step >
-    # current_step, it means a model reset happens while the simulation is
-    # still playing.
-    previous_step = solara.use_reactive(0)
-
-    def on_value_play(change):
-        if previous_step.value > current_step and current_step == 0:
-            # We add extra checks for current_step == 0, just to be sure.
-            # We automatically stop the playing if a model is reset.
-            playing.value = False
-        elif model.running:
-            do_step()
-        else:
-            playing.value = False
 
-    def do_step():
-        model.step()
-        previous_step.value = current_step
-        set_current_step(model.schedule.steps)
+    def on_value_play(_):
+        if playing.value:
+            on_step()
+
+    def reset():
+        playing.value = False
+        on_reset()
+
+    with solara.Card(), solara.Row(gap="2px", style={"align-items": "center"}):
+        with solara.Tooltip("Reset the model"):
+            solara.Button(icon_name="mdi-reload", color="primary", on_click=reset)
+
+        with solara.Tooltip("Step forward"):
+            solara.Button(label="▶|", color="primary", on_click=on_step)
```

Comment:
This symbol is not in https://en.wikipedia.org/wiki/Media_control_symbols. I think `+1` or `-1` (for step backward) is clearer?